packages/ember-htmlbars/tests/helpers/if_unless_test.js

```diff
@@ -51,248 +47,6 @@ QUnit.module('ember-htmlbars: {{#if}} and {{#unless}} helpers', {
-  });
-
-  runAppend(view);
-
-  equal(view.$().text(), 'YepBoundYepUnbound');
-});
-
-QUnit.test('The `if` helper does not print the contents for an object proxy without content', function() {
-  view = EmberView.create({
-    truthy: ObjectProxy.create({ content: {} }),
-    falsy: ObjectProxy.create({ content: null }),
-
-    template: compile('{{#if view.truthy}}Yep{{/if}}{{#if view.falsy}}Nope{{/if}}')
-  });
-
-  runAppend(view);
-
-  equal(view.$().text(), 'Yep');
-});
-
-QUnit.test('The `if` helper updates if an object proxy gains or loses context', function() {
-  view = EmberView.create({
-    proxy: ObjectProxy.create({ content: null }),
-
-    template: compile('{{#if view.proxy}}Yep{{/if}}')
-  });
-
-  runAppend(view);
-
-  equal(view.$().text(), '');
-
-  run(function() {
-    view.set('proxy.content', {});
-  });
-
-  equal(view.$().text(), 'Yep');
-
-  run(function() {
-    view.set('proxy.content', null);
-  });
-
-  equal(view.$().text(), '');
-});
-
-function testIfArray(array) {
-  view = EmberView.create({
-    array: array,
-
-    template: compile('{{#if view.array}}Yep{{/if}}')
-  });
-
-  runAppend(view);
-
-  equal(view.$().text(), '');
-
-  run(function() {
-    view.get('array').pushObject(1);
-  });
-
-  equal(view.$().text(), 'Yep');
-
-  run(function() {
-    view.get('array').removeObject(1);
-  });
-
-  equal(view.$().text(), '');
-}
-
-QUnit.test('The `if` helper updates if an array is empty or not', function() {
-  testIfArray(emberA());
-});
-
-QUnit.test('The `if` helper updates if an array-like object is empty or not', function() {
-  testIfArray(ArrayProxy.create({ content: emberA() }));
-});
-
-QUnit.test('The `unless` helper updates if an array-like object is empty or not', function() {
-  view = EmberView.create({
-    array: ArrayProxy.create({ content: emberA() }),
-
-    template: compile('{{#unless view.array}}Yep{{/unless}}')
-  });
-
-  runAppend(view);
-
-  equal(view.$().text(), 'Yep');
-
-  run(function() {
-    view.get('array').pushObject(1);
-  });
-
-  equal(view.$().text(), '');
-
-  run(function() {
-    view.get('array').removeObject(1);
-  });
-
-  equal(view.$().text(), 'Yep');
-});
-
-QUnit.test('The `if` helper updates when the value changes', function() {
-  view = EmberView.create({
-    conditional: true,
-    template: compile('{{#if view.conditional}}Yep{{/if}}')
-  });
-  runAppend(view);
-  equal(view.$().text(), 'Yep');
-  run(function() {
-    view.set('conditional', false);
-  });
-  equal(view.$().text(), '');
-});
-
-QUnit.test('The `if (unbound` helper does not update when the value changes', function() {
-  view = EmberView.create({
-    conditional: true,
-    template: compile('{{#if (unbound view.conditional)}}Yep{{/if}}')
-  });
-  runAppend(view);
-  equal(view.$().text(), 'Yep');
-  run(function() {
-    view.set('conditional', false);
-  });
-  equal(view.$().text(), 'Yep');
-});
-
-QUnit.test('The `unless` helper updates when the value changes', function() {
```

Comment:
Replaced by:

* BASIC_TRUTHY_TESTS
* BASIC_FALSY_TESTS

(Now testing re-render, mutation and replacement as well)